### ISSUE TITLE
Unauthorized when trying to get /catalog/v0/items while getting /orders/v0/orders works fine

### ISSUE THREAD

**Issue by kwimbeep** [amzn/selling-partner-api-docs#397]
Greetings, /catalog/v0/items returns "Access to requested resource is denied" error.
Getting orders and /products/pricing/v0/price works ok.

At first getting orders did not work as well, but after AssumeRole it works now.
For item listing I'm using the same tokens, same marketplace etc, but still getting the error.

App is using self authorization with role as IAM ARN. In App Settings page I have checked these roles: Finance and Accounting, Pricing, Inventory and Order Tracking, Amazon Fulfillment and Product Listing.

What could be the problem? Does anyone have any idea?

**Comment by ShivikaK** [comment]
Hello,

Thank you for raising this issue.

Please open a support case with us as we would like to review your application permissions and request details to further troubleshoot the issue.

Thanks,
Shivika Khare
Selling Partner API Developer Support

**Comment by educatornum** [comment]
Hello,
In my case, **Fulfillment Inbound**  and **FeedAPI** were ok, but the **Catalog Items** doesn't work. 
Then this page [sellercentral](https://sellercentral-japan.amazon.com/home) I have checked these roles but I received the same error with @kwimbeep. 

**Here is my  SellerCentrel's setting  role:**
![Screen Shot 2021-02-16 at 10 48 16](https://user-images.githubusercontent.com/25072337/108008836-9e3fcb00-7044-11eb-9e7e-3a9034e9b73f.png)

there is still an error...
```
{
      'code': 'Unauthorized',
      'details': '',
      'message': 'Access to requested resource is denied.'
}
```

What could be the problem? Is it a problem with the **IAM ARN permission?**
Does anyone have any idea?

Many thanks,

**Comment by johnkw** [comment]
We experienced this same bug. It was fixed by going to the https://sellercentral.amazon.com/sellingpartner/developerconsole/ page, then "Edit app", and then checking all 6 possible checkboxes. I don't know which checkbox fixed it. Seller Support never responded with anything useful - they seem to have no idea what caused the problem or what fixed it.

**Comment by educatornum** [comment]
Did somebody solve the problem?

**Comment by estebanangel** [comment]
I only see 3 checkboxes in the developer profile in Seller Central, any idea how to get the 6 checkboxes?

**Comment by educatornum** [comment]
@estebanangel

**Comment by estebanangel** [comment]
> @estebanangel
> What is your marketplace?

North America

**Comment by ShivikaK** [comment]
Hello @estebanangel 

The checkboxes that appear on "Edit App" page are based on the selection you made when you completed Developer Profile form. 

You can review the Developer Profile and select all the roles that you require access to including Restricted roles. 

Once the form is approved, you can then see those roles in Edit App page and select all the roles you need access to for SP API and save the app with the changes made.

Thanks,
Shivika Khare
Selling Partner API Developer Support

**Comment by educatornum** [comment]
@ShivikaK 
Hello, 
How about it? Could you support the problem?

> In my case, **Fulfillment Inbound** and **FeedAPI** were ok, but the **Catalog Items** don't work.
> Then this page [sellercentral](https://sellercentral-japan.amazon.com/home) I have checked these roles but I received the same error with @kwimbeep.
> 
> **Here is my SellerCentrel's setting role:**
> ![Screen Shot 2021-02-16 at 10 48 16](https://user-images.githubusercontent.com/25072337/108008836-9e3fcb00-7044-11eb-9e7e-3a9034e9b73f.png)
> 
> there is still an error...
> 
> ```
> {
>       'code': 'Unauthorized',
>       'details': '',
>       'message': 'Access to requested resource is denied.'
> }
> ```
> 
> What could be the problem? Is it a problem with the **IAM ARN permission?**
> Does anyone have any idea?
> 
> Many thanks,

**Comment by ShivikaK** [comment]
Hello @educatornum 

Please review if any of the below steps help troubleshoot the issue -

1. If the app has new roles added to it, ensure that the app has been re-authorized to get access to that API i.e. new LWA refresh token has been generated.
2. Verify IAM entity associated with the application. For IAM user, IAM policy needs to be directly attached to the user. For IAM role, you need to attach policy to role and use AWS STS to request temporary credentials along with session token which needs to be added to your SP API request.
3. If there is a support case reported for this issue, please let me know and I can get it expedited for further assistance.

Thanks,
Shivika Khare
Selling Partner API Developer Support

**Comment by estebanangel** [comment]
> Hello @estebanangel
> 
> The checkboxes that appear on "Edit App" page are based on the selection you made when you completed Developer Profile form.
> 
> You can review the Developer Profile and select all the roles that you require access to including Restricted roles.
> 
> Once the form is approved, you can then see those roles in Edit App page and select all the roles you need access to for SP API and save the app with the changes made.
> 
> Thanks,
> Shivika Khare
> Selling Partner API Developer Support

This worked, thank you very much. I was looking initially at the App settings and I could only see the 3 checkboxes.

**Comment by johnkw** [comment]
I think I've mentioned this elsewhere in one of these github bugs, but it bears repeating: If the User is missing a checkbox, or the App is missing a checkbox on the https://sellercentral.amazon.com/sellingpartner/developerconsole/  page like I described above, the SP-API error code needs to say that SPECIFICALLY.  It needs to say the EXACT checkbox that is missing, not just "Access to requested resource is denied", which is almost completely meaningless and useless.

**Comment by educatornum** [comment]
> Hello @educatornum
> 
> Please review if any of the below steps help troubleshoot the issue -
> 
> 1. If the app has new roles added to it, ensure that the app has been re-authorized to get access to that API i.e. new LWA refresh token has been generated.
> 2. Verify IAM entity associated with the application. For IAM user, IAM policy needs to be directly attached to the user. For IAM role, you need to attach policy to role and use AWS STS to request temporary credentials along with session token which needs to be added to your SP API request.
> 3. If there is a support case reported for this issue, please let me know and I can get it expedited for further assistance.
> 
> Thanks,
> Shivika Khare
> Selling Partner API Developer Support

@ShivikaK

**Comment by estebanangel** [comment]
Getting back to his issue, I'm getting `'403 Client Error: Forbidden for url: https://sellingpartnerapi-na.amazon.com/reports/2020-09-04/reports` when trying to get the `GET_FLAT_FILE_RETURNS_DATA_BY_RETURN_DATE`. Do someone have access to this report? I'm getting access to all the orders, inventory, finances reports. Thanks

**Comment by johnkw** [comment]
See bug #371 for the lack of documentation for reports permissions.

**Comment by quazardous** [comment]
I dont' know if it's related but I'm having a strange behaviour with [getpricing](https://github.com/amzn/selling-partner-api-docs/blob/main/references/product-pricing-api/productPricingV0.md#getpricing) vs [getitemoffers](https://github.com/amzn/selling-partner-api-docs/blob/main/references/product-pricing-api/productPricingV0.md#getitemoffers).

getitemoffers is working perfectly but not getpricing !?

with `GET https://sellingpartnerapi-eu.amazon.com/products/pricing/v0/price?Asins=2070584623&ItemType=ASIN&MarketplaceId=A13V1IB3VIYZZH`

I Get a 404
```json
{
      "code": "NotFound",
      "message": "The requested resource doesn\u0027t exist."
}
```
Any clue ?

**Comment by AmandeepSingh179** [comment]
> Getting back to his issue, I'm getting `'403 Client Error: Forbidden for url: https://sellingpartnerapi-na.amazon.com/reports/2020-09-04/reports` when trying to get the `GET_FLAT_FILE_RETURNS_DATA_BY_RETURN_DATE`. Do someone have access to this report? Should we check all the restricted roles to get access to this report? I'm getting access to all the orders, inventory, finances reports. Thanks

Hi, 
Were you able to sort out this issue. I'm facing the same issue> I can download `GET_MERCHANT_LISTINGS_ALL_DATA` and `GET_FLAT_FILE_ALL_ORDERS_DATA_BY_ORDER_DATE_GENERAL`, but `GET_FLAT_FILE_RETURNS_DATA_BY_RETURN_DATE` is throwing 403 `Access to the resource is forbidden` response.

**Comment by ouroboros25** [comment]
@quazardous  Hello! Did you solve a problem with The requested resource doesn\u0027t exist?

**Comment by quazardous** [comment]
@ouroboros25 yes it was an input typo: I spelled ASIN but it was Asin...

**Comment by vovarpd** [comment]
> Hello @educatornum
> 
> Please review if any of the below steps help troubleshoot the issue -
> 
> 1. If the app has new roles added to it, ensure that the app has been re-authorized to get access to that API i.e. new LWA refresh token has been generated.
> 2. Verify IAM entity associated with the application. For IAM user, IAM policy needs to be directly attached to the user. For IAM role, you need to attach policy to role and use AWS STS to request temporary credentials along with session token which needs to be added to your SP API request.
> 3. If there is a support case reported for this issue, please let me know and I can get it expedited for further assistance.
> 
> Thanks,
> Shivika Khare
> Selling Partner API Developer Support

Thanks. 

I spent a lot of time debugging 403 errors (checkboxes were checked).
Finally, I created a new app, with a new LWA, with a new refresh_token and it works fine.

Looks like editing permission on the existing app just not working.

**Comment by QasimMughal05** [comment]
> Asin

How did you solve it?

**Comment by QasimMughal05** [comment]
> @ouroboros25 yes it was an input typo: I spelled ASIN but it was Asin...

How did you solve it?

**Comment by adillucky** [comment]
https://sellingpartnerapi-eu.amazon.com/orders/v0/orders?CreatedAfter=2021-04-27T16:20:44Z&MarketplaceIds=AKOAFLDV1D5DL

{
"payload": {
"Orders": [],
"CreatedBefore": "2022-04-27T16:26:45Z"
}
}

i m also facing this type of issue i have many orders in seller center but when i try to get order according to order ID its working but when i try to get all orders i m facing this issue.

**Comment by johnkw** [comment]
Pretty sure this is still broken, unless there was another bug opened for indecipherable error messages on permission problems.

**Comment by hellfire989** [comment]
> Pretty sure this is still broken, unless there was another bug opened for indecipherable error messages on permission problems.

I'm still getting this issue. Is there an updated fix?

**Comment by Peymanpn** [comment]
I am having the same issue with the specific report 
`GET_SALES_AND_TRAFFIC_REPORT` and `GET_BRAND_ANALYTICS_MARKET_BASKET_REPORT`
 it was working fine until the last week.

do I need to open a new ticket for this or this one covers it?

reponse:
```json
{"code": "Unauthorized", "message": "Access to the resource is forbidden", "details": "" }
```

**Comment by hellfire989** [comment]
> this

I actually fixed the issue. You need to make sure your app has permissions for "Products" AND "Product Pricing". 

This is set up when you first registered your developer account.

Your account can be revised to include this.
